pkg/netceptor/conn.go

```diff
@@ -31,6 +31,47 @@ var MaxIdleTimeoutForQuicConnections = 30 * time.Second
 // Having this variablized allows the tests to set KeepAliveForQuicConnections = False so that things will properly fail.
 var KeepAliveForQuicConnections = true
 
+// quicListenerAdapter adapts *quic.Listener to QuicListenerForListener interface.
+// This allows real QUIC listeners to work with our interface while enabling test mocking.
+type quicListenerAdapter struct {
+	*quic.Listener
+}
+
+func (a *quicListenerAdapter) Accept(ctx context.Context) (QuicConnectionForConn, error) {
+	conn, err := a.Listener.Accept(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to accept connection: %w", err)
+	}
+	if conn == nil {
```

Comment:
Looking at the quic-go code, if err == nil conn is defined. This check is redundant.

Reference:  https://github.com/quic-go/quic-go/blob/master/server.go#L333